File: ExerciceGTI/src/main/java/com/GTI/ExerciceGTI/dataTransferObjects/DemandeCreditRequest.java
package com.GTI.ExerciceGTI.dataTransferObjects;


import lombok.*;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
@Getter
@Setter
@Builder
public class DemandeCreditRequest {

    private Integer ncin;
    private Integer ncompte;
    private String observation;
    private Integer type;
    private Integer unite;
    private Integer nbreEcheance;
    private Integer montant;
    private double taux;
    private List<GarantieRequest> garantieRequests;

}

File: eCredit/src/app/interfaces/DemandeCreditRequest.ts
import { Garantie } from './Garantie';
export interface DemandeCreditRequest {
    numDemande?:number;
    ncin:number;
    ncompte:number;
    type:number;
    montant:number;
    unite:number;
    nbreEcheance:number;
    observation:string;
    taux?:number;
    garantieRequests:Garantie[];
}